glue/core/component.py

```diff
@@ -64,10 +68,20 @@ def units(self, value):
         else:
             self._units = str(value)
```

Comment:
```suggestion
        else:
            new_units = str(value)
        if getattr(self, '_original_units', '') == '':
            self._original_units = new_units
        if not u.Unit(self._original_units).is_equivalent(u.Unit(new_units)):
            raise u.UnitConversionError(f"New unit '{new_units}' must be convertible "
                                        f"from original one '{self._original_units}'")
        self._units = new_units
```
Doh, that should also include
```python
        if value is None:
            new_units = ''
```
but GH won't let me touch that line...
Anyway, proposing here to allow setting a novel unit type _once_ from a unitless/dimensionless component, and then raising immediately on non-equivalent units so they cannot be set in the first place (otherwise will raise later in `__getitem__` when the damage is already done.